services/adapter/src/main/java/tech/pegasys/artemis/services/adapter/ServiceAdapter.java

```diff
@@ -1,112 +0,0 @@
-import tech.pegasys.artemis.services.adapter.event.OutboundEvent;
-import tech.pegasys.artemis.services.adapter.factory.MethodDescriptorFactory;
-import tech.pegasys.artemis.services.adapter.io.inbound.DefaultGrpcServer;
-import tech.pegasys.artemis.services.adapter.io.inbound.GrpcServer;
-import tech.pegasys.artemis.services.adapter.io.outbound.EventForwarder;
-import tech.pegasys.artemis.services.adapter.io.outbound.GrpcEventForwarder;
-
-/** Encapsulates receiving/delivering events from/to Artemis microservices */
-public class ServiceAdapter implements ServiceInterface {
-
-  public static final String SERVICE_NAME = "tech.pegasys.artemis.serviceAdapter";
-
-  private GrpcServer server;
-
-  private Set<MethodDescriptor<?, RemoteCallResponse>> inboundDescriptors;
-
-  private Set<EventForwarder<?>> eventForwarders;
-
-  private int serverPort;
-
-  private EventBus eventBus;
-
-  private boolean hasRegisteredInboundEvents = false;
-
-  public ServiceAdapter(
-      int serverPort, Set<Class<?>> inboundEvents, Set<OutboundEvent<?>> outboundEvents) {
-    this.serverPort = serverPort;
-    this.eventForwarders = new HashSet<>();
-    this.inboundDescriptors = new HashSet<>();
-
-    if (inboundEvents != null && !inboundEvents.isEmpty()) {
-      hasRegisteredInboundEvents = true;
-    }
-
-    server = createGrpcServer(serverPort);
-
-    inboundEvents.forEach(
-        inboundEvent -> server.registerMethodDescriptor(createMethodDescriptor(inboundEvent)));
-
-    outboundEvents.forEach(outboundEvent -> registerEventForwarder(outboundEvent));
-  }
-
-  public ServiceAdapter(Set<Class<?>> inboundEvents, Set<OutboundEvent<?>> outboundEvents) {
-    this(0, inboundEvents, outboundEvents);
-  }
-
-  @Override
-  public void init(ServiceConfig config) {
-    this.eventBus = config.getEventBus();
-    eventForwarders.forEach(forwarder -> forwarder.init(eventBus));
-  }
-
-  @Override
-  public void run() {
-    server.run();
-  }
-
-  @Override
-  public void stop() {
-    server.stop();
-
-    eventForwarders.forEach(EventForwarder::stop);
-  }
-
-  protected MethodDescriptor<?, RemoteCallResponse> createMethodDescriptor(Class<?> eventClass) {
-    return MethodDescriptorFactory.build(SERVICE_NAME, eventClass);
-  }
-
-  protected GrpcServer createGrpcServer(int serverPort) {
-    return new DefaultGrpcServer(SERVICE_NAME, serverPort, this::onInboundEvent);
-  }
-
-  private void onInboundEvent(Object event) {
-    eventBus.post(event);
-  }
-
-  private void registerEventForwarder(OutboundEvent<?> outboundEvent) {
-    final Channel channel =
-        ManagedChannelBuilder.forTarget(outboundEvent.getUrl()).usePlaintext().build();
-
-    final MethodDescriptor<?, RemoteCallResponse> descriptor =
-        MethodDescriptorFactory.build(SERVICE_NAME, outboundEvent.getEventClass());
-
-    final EventForwarder<?> forwarder = new GrpcEventForwarder<>(channel, descriptor);
-
-    eventForwarders.add(forwarder);
-  }
-}
```

Comment:
Don't you think this file could be useful in the future?

Reply:
If it is, we can always pull it out of history.  Most likely though if we do move to a multiprocess architecture we should just use an event bus implementation that supports that rather than having point to point repeaters in-place.